### PR TITLE
Feature login

### PR DESCRIPTION
**Pull request by FrankIglesias**
# Summary
- Added a modal window for login
- The logged status keeps persistent 
- The user is not able to go to any page without sign in first

# Trello card
https://trello.com/c/9PSNhFA3/21-login

# Screenshot
screenshot from 2017-06-21 11-39-48

# Notes 
in order to sign in you can use this:
 email: react.dev@wolox.com.ar
 password: 12345678

### CODE REVIEW

File: my-app/src/App.css [hunk 1]
```diff
@@ -193,3 +193,47 @@
...
+  margin: 10px 0; }
+
+.cancel-button {
+  background: #CCC;
+  border-radius: 4px;
+  border: 0;
+  color: white;
+  height: 44px;
+  margin: 0 30px;
+  width: 189px; }
```
Comment (pabloferro) [inline my-app/src/App.css]: If you have style in common you can add a new class like `button`.
And then the components will have `className="button cancel-button"` for example.

File: my-app/src/modules/App/Modal.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      passwordError:false,
+      passwordMessage: '',
+      emailMessage: ''
+    }
+  }
+
+handleChangeMail = (e) =>{this.setState({email: e.target.value})}
+handleChangePass = (e)=> {this.setState({password: e.target.value})}
+
+Login(){
```
Comment (pabloferro) [inline my-app/src/modules/App/Modal.js]: lowercase for functions.

File: my-app/src/modules/App/Modal.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  axios.post('https://wbooks-api-stage.herokuapp.com/api/v1/users/sessions', {
+    email: this.state.email,
+    password: this.state.password
+  }).then( (response)=> {
+    if(response.status ==200){
+      localStorage.setItem('logged',true)
+      this.setState({logged:true})
```
Comment (pabloferro) [inline my-app/src/modules/App/Modal.js]: You'll probably need to store the access token. Is it done in another pull request?

File: my-app/src/modules/App/Modal.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    this.setState({logginError:true})
+  })
+}
+
+submit = (e) => {
+  e.preventDefault();
+    this.setState({emailError:false,
+              passwordError:false})
+     var password =this.state.password;
+     var email =this.state.email;
```
Comment (pabloferro) [inline my-app/src/modules/App/Modal.js]: setState is asynchronous, but just for clarity I'd reset the state after creating the constants.

Also use always `const` or `let` (`const` here).

File: my-app/src/modules/App/Modal.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+     if(password.length< 8 || password.length > 52){
+       this.setState({passwordError:true,
+                      passwordMessage: 'La cantidad de caracteres debe ser menor a 8 o mayor a 52'})
+     }else if(!(/\d/).test(password)){
+       this.setState({passwordError:true,
+                      passwordMessage: 'La contraseña debe contener al menos un numero y una letra'})
+     }else if(!email.includes('@') || !email.includes('.')){
+       this.setState({emailError: true})
```
Comment (pabloferro) [inline my-app/src/modules/App/Modal.js]: Using else ifs you'll only show one error at the time.
One way to do this would be to have  an array of errors (for each field if you want) and push the messages when they are needed.

File: my-app/src/modules/App/Modal.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+     }
+      else{
+        this.Login()
+     }
+}
+
+render(){
+return (
+  <div className="modal-window">
+   {this.state.logged ? <Redirect to="/dashboard" push/> : <ModalDialog>
```
Comment (pabloferro) [inline my-app/src/modules/App/Modal.js]: Here you can use an if in the render like:
```js
render(){
  if (this.state.logged) {
    return (<Redirect to="/dashboard" push/>);
  }
  
  return (
    // the modal
  );
}
```

File: my-app/src/modules/Book/Book.js [hunk 7]
```diff
@@ -9,6 +9,11 @@ import UserComment from './UserComment.js'
 
 class Book extends Component {
   render(){
+
+    if(localStorage.getItem('logged')===null){
```
Comment (pabloferro) [inline my-app/src/modules/Book/Book.js]: `if (!localStorage.getItem('logged')) {`

File: my-app/src/modules/Home.js [hunk 8]
```diff
@@ -46,33 +46,40 @@ class Home extends Component {
   }
 
   render() {
+    if(localStorage.getItem('logged')===null){
```
Comment (pabloferro) [inline my-app/src/modules/Home.js]: same

File: my-app/src/modules/Book/Book.js (hunk 7 above)
Comment (pabloferro) [inline my-app/src/modules/Book/Book.js]: Also try to find a way of not repeating this part.
You may need to use High Order Components or just do something in the routes.

File: my-app/src/modules/App/Modal.js (hunk 3 above)
Comment (FrankIglesias) [inline my-app/src/modules/App/Modal.js]: It is !
